adapter/ipListChecker/ipListChecker.go

```diff
@@ -89,27 +91,52 @@ func (builder) ValidateConfig(cfg adapter.AspectConfig) (ce *adapter.ConfigError
 }
 
 func newListChecker(env adapter.Env, c *config.Params) (*listChecker, error) {
+	return newListCheckerWithTimers(env, c,
+		time.NewTicker(time.Second*time.Duration(c.RefreshInterval)),
+		time.NewTimer(time.Second*time.Duration(c.Ttl)))
+}
+
+func newListCheckerWithTimers(env adapter.Env, c *config.Params,
+	refreshTicker *time.Ticker, purgeTimer *time.Timer) (*listChecker, error) {
 	l := &listChecker{
-		log:             env.Logger(),
-		providerURL:     c.ProviderUrl,
-		closing:         make(chan bool),
-		refreshInterval: time.Second * time.Duration(c.RefreshInterval),
-		ttl:             time.Second * time.Duration(c.Ttl),
+		log:           env.Logger(),
+		providerURL:   c.ProviderUrl,
+		refreshTicker: refreshTicker,
+		purgeTimer:    purgeTimer,
+		ttl:           time.Second * time.Duration(c.Ttl),
 	}
-	l.client = http.Client{Timeout: l.ttl}
-	l.setList(listState{})
+	l.setListState(listState{})
 
 	// load up the list synchronously so we're ready to accept traffic immediately
-	l.refreshList()
-
-	// crank up the async list refresher
-	go l.listRefresher()
+	_ = l.fetchList()
```

Comment:
Not particularly. The problem will be logged, like other failures. I'll add a comment to that effect.